cirq/ops/pauli.py

```diff
@@ -39,6 +39,14 @@ def third(self, second: 'Pauli') -> 'Pauli':
     def difference(self, second: 'Pauli') -> int:
         return (self._index - second._index + 1) % 3 - 1
 
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._index == other._index
+
+    def __ne__(self, other):
```

Comment:
Are you sure that this will also return NotImplemented for the objects of different types? Is there any test to verify that?

Reply:
This is the normal way to do it.  If other is a different type, it will use `other.__eq__` and either return a bool or throw a TypeError.